mutation_partition.cc

```diff
@@ -305,14 +305,104 @@ stop_iteration mutation_partition::apply_monotonically(const schema& s, mutation
     _static_row.apply_monotonically(s, column_kind::static_column, std::move(p._static_row));
     _static_row_continuous |= p._static_row_continuous;
 
+    rows_entry::tri_compare cmp(s);
+    auto del = current_deleter<rows_entry>();
+
+    // Compacts rows in [i, end) with the tombstone.
+    // Erases entries which are left empty by compaction.
+    // Does not affect continuity.
+    auto apply_tombstone_to_rows = [&] (apply_resume::stage stage, tombstone tomb, rows_type::iterator i, rows_type::iterator end) -> stop_iteration {
```

Comment:
Can we solve #2577 in the same way? Instead of apply_tombstones_to_rows we apply_schema_upgrade_to_rows, but otherwise same.

Reply:
I think it's simpler not to. We can maintain the invariant that each version conforms to a single schema. Then upgrading schema will have to be done by the means of moving atoms from one version to another. The only change we need to do here is to teach the algorithm to upgrade schema of the moved elements.